### PR TITLE
truly optional rocksdb

### PR DESCRIPTION
**Pull request by drahnr**
## What does it do?
 Currently the `sc-client-db` is pulled in with default features at the level `cli` which makes it a defacto dependency for all parachains. With the latest slew of compiler releases not include `<cstdint>` by default anymore, and `rocksdb` binding a significant amount of CI resources, I felt it was time to get rid of it without having to continually patch substrate.

## What important points should reviewers know?

RocksDB is now optional two levels further up, behind another `rocksdb` feature flag defaulting to on. The default behavior should not have changed.

## Is there something left for follow-up PRs?

`node-testing` makes some assumptions of having some identifiers around which are not upheld, it surfaced when testing `no-default-features`, but otherwise no change.

---

We are successfully using this with `polkadot-v0.9.41` (plus a few patches) and clang-16 and gcc-13.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate-macos
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2813378

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2813432

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2813431

**Comment by niklasad1** [comment]
bot rebase

File: bin/node/cli/Cargo.toml [hunk 1]
```diff
@@ -100,14 +100,14 @@ node-executor = { version = "3.0.0-dev", path = "../executor" }
...
-frame-benchmarking-cli = { version = "4.0.0-dev", optional = true, path = "../../../utils/frame/benchmarking-cli" }
+frame-benchmarking-cli = { version = "4.0.0-dev", default-features = false, optional = true, path = "../../../utils/frame/benchmarking-cli" }
 node-inspect = { version = "0.9.0-dev", optional = true, path = "../inspect" }
 try-runtime-cli = { version = "0.10.0-dev", optional = true, path = "../../../utils/frame/try-runtime/cli" }
 serde_json = "1.0.85"
 
 [dev-dependencies]
 sc-keystore = { version = "4.0.0-dev", path = "../../../client/keystore" }
-sc-client-db = { version = "0.10.0-dev", path = "../../../client/db" }
+sc-client-db = { version = "0.10.0-dev", default-features=false, path = "../../../client/db" }
```
Comment (niklasad1) [inline bin/node/cli/Cargo.toml:110]:
```suggestion
sc-client-db = { version = "0.10.0-dev", default-features = false, path = "../../../client/db" }
```

**Review by niklasad1 — APPROVED** [review]
nice, looks good

**Comment by drahnr** [comment]
Are the `cumulus` and `polkadot` CI checks caused by the changeset or are they known issues?

Could I get another review :)

**Comment by niklasad1** [comment]
> Are the cumulus and polkadot CI checks caused by the changeset or are they known issues?

It's probably just because there were a bunch PRs merged recently that broke that, rebased the branch and it's seems to be okay but cumulus is currently broken because https://github.com/paritytech/cumulus/pull/2574

So nothing to worry about

**Comment by bkchr** [comment]
> and `rocksdb` binding a significant amount of CI resources

`ROCKSDB_LIB_DIR="${rocksdb}/lib";` solves this problem.

And why are you using `node-cli` in your parachain?

**Comment by drahnr** [comment]
> > and `rocksdb` binding a significant amount of CI resources
> 
> `ROCKSDB_LIB_DIR="${rocksdb}/lib";` solves this problem.

I am aware, which was my [first quarrel with](https://github.com/rust-rocksdb/rust-rocksdb/pull/694
) it. Now we had the gcc/clang update as part of our build container and again rocksdb made em all red. 
> And why are you using `node-cli` in your parachain?

We are not. `sc-cli` and `sc-client` are used, let me double check if the other changes are already part of `master` and were just missing in the `0.9.4x` branch we use.

**Pull request closed without merging**